evm/src/main/java/org/hyperledger/besu/evm/worldstate/JournaledAccount.java

```diff
@@ -349,13 +393,118 @@ public void setStorageWasCleared(final boolean storageWasCleared) {
     if (immutable) {
       throw new ModificationNotAllowedException();
     }
-    this.storageWasCleared = storageWasCleared;
+    this.storageWasCleared.set(storageWasCleared);
+  }
+
+  /**
+   * Create an immutable snapshot of this account at the provided mark.
+   *
+   * @param mark the mark to snapshot at
+   * @return an {@link MutableAccount} representing the state at {@code mark}
+   */
+  public Account snapshot(final long mark) {
+    final long snapNonce = nonce.get(mark);
+    final Wei snapBalance = balance.get(mark);
+    final Bytes snapCode = code.get(mark);
+    final Hash snapCodeHash = codeHash.get(mark);
+    final boolean snapStorageCleared = storageWasCleared.get(mark);
+    final NavigableMap<UInt256, UInt256> snapStorage = new TreeMap<>(updatedStorage.toMap(mark));
+
+    final Account wrappedSnapshot;
+    if (account instanceof JournaledAccount ja) {
+      wrappedSnapshot = ja.snapshot(mark);
+    } else {
+      wrappedSnapshot = account;
```

Comment:
what about throwing an error here? I think if i was trying to get a snapshot(markX) without realizing the underlying implementation didn't support marking, i'd be surprised to always get whatever the current value of account is, the parameter getting ignored.

Unless this only occurs for the first mark since the journaling chain hasn't started yet?